scalikejdbc-core/src/main/scala/scalikejdbc/DBSession.scala

```diff
@@ -247,7 +249,21 @@ trait DBSession extends LogSupport with LoanPattern {
    * @return result as list
    */
   def list[A](template: String, params: Any*)(extract: WrappedResultSet => A): List[A] = {
-    traversable(template, params: _*)(extract).toList
+    collection[A, List](template, params: _*)(extract)
+  }
+
+  /**
+   * Returns query result as any Collection object.
+   *
+   * @param template SQL template
+   * @param params parameters
+   * @param extract extract function
+   * @tparam A return type
+   * @tparam C[_] return collection type
```

Comment:
I can not find the scaladoc spec of the higher kinded tparam.
I'll try generate scaladoc and check the behavior at home.